gobblin-compaction/src/main/java/org/apache/gobblin/compaction/mapreduce/CompactionAvroJobConfigurator.java

```diff
@@ -270,7 +278,10 @@ public Job createJob(FileSystemDataset dataset) throws IOException {
     addJars(conf);
     Job job = Job.getInstance(conf);
     job.setJobName(MRCompactorJobRunner.HADOOP_JOB_NAME);
-    this.configureInputAndOutputPaths(job, dataset);
+    if (!this.configureInputAndOutputPaths(job, dataset)) {
+      log.info("No valid input paths found for dataset: " + dataset.getUrn());
+      return null;
```

Comment:
It's worth considering whether we want to create an empty output directory in this case, since that is technically the correct compacted partition, and it will not impact consumers expecting the partition to be present.

Reply:
Good point! Addressed it. Thanks!